### DISCUSSION TITLE
Can I enable coercion by default?

### DISCUSSION THREAD

**Discussion by NullEnt1ty** [colinhacks/zod#1910] [Q&A] [answered]
I'm using `yup` at the moment where I can call `MySchema.validate(input)` to parse and cast the input. When building the same schema with `zod` I would have to use `z.coerce` for every primitive type in order to achieve the same result.

Is there any way to enable coercion for primitives by default?

**Comment by JacobWeisenburger** [comment, marked as answer]
Currently, I don't think there is a way to do this. You would need to create a utility function for this.

**Reply by NullEnt1ty** [reply]
Gotcha, thank you!
